internal/perses/common/perses_client_factory.go

```diff
@@ -19,10 +21,21 @@ func NewWithConfig() PersesClientFactory {
 }
 
 func (f *PersesClientFactoryWithConfig) CreateClient(perses persesv1alpha1.Perses) (v1.ClientInterface, error) {
-	restClient, err := perseshttp.NewFromConfig(perseshttp.RestConfigClient{
-		URL: fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", perses.Name, perses.Namespace, perses.Spec.ContainerPort),
-	})
+	var urlStr string
+
+	if flag.Lookup("perses-server-url").Value.String() != "" {
+		urlStr = flag.Lookup("perses-server-url").Value.String()
+	} else {
+		urlStr = fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", perses.Name, perses.Namespace, perses.Spec.ContainerPort)
+	}
+	parsedURL, err := common.ParseURL(urlStr)
+	if err != nil {
+		return nil, err
+	}
 
+	restClient, err := clientConfig.NewRESTClient(clientConfig.RestConfigClient{
+		URL: &common.URL{URL: parsedURL.URL},
```

Comment:
```suggestion
URL: parsedURL,
```

Reply:
fixed in https://github.com/perses/perses-operator/pull/21/commits/3849b48dbf6df3e9a4f1392cdb7065acc248aa43 :)